.github/workflows/node.js.yml

```diff
@@ -1,34 +1,64 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+
 # This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
 
 name: Node.js CI
-
+run-name: ${{ github.actor }} running CI suite
 on:
   push:
     branches:
       - master
       - test
+      - 448-improve-GH-actions #TODO: remove this
   pull_request:
     branches:
       - master
       - test
-
 jobs:
-  build:
-
+  CI-tests:
```

Comment:
Can we:
1) treat linting+type checking and building+testing as two different jobs
2) Label steps with names

See [ECharts GH actions](https://github.com/apache/echarts/blob/master/.github/workflows/ci.yml), for example.